File: import-posts.js
const fs = require('fs');
const path = require('path');
const { parse } = require('csv-parse/sync');

// Read the CSV file
const csvPath = '/Users/kimberlybella/Downloads/Daily(ish) GenAI Newsletter - Sheet1.csv';
const csvContent = fs.readFileSync(csvPath, 'utf-8');

// Parse CSV properly with csv-parse library
const records = parse(csvContent, {
  columns: true,
  skip_empty_lines: true,
  relax_quotes: true,
});

console.log(`Found ${records.length} records to process`);

// Helper to create slug from title
function createSlug(title) {
  return title
    .toLowerCase()
    .replace(/[^a-z0-9]+/g, '-')
    .replace(/^-+|-+$/g, '');
}

// Helper to format date
function formatDate(dateStr) {
  if (!dateStr) return new Date().toISOString().split('T')[0];

  const parts = dateStr.split('/');
  if (parts.length === 3) {
    const month = parts[0].padStart(2, '0');
    const day = parts[1].padStart(2, '0');
    const year = parts[2];
    return `${year}-${month}-${day}`;
  }
  return dateStr;
}

// Process each row
let postsCreated = 0;

for (const row of records) {
  const title = row['Title']?.trim();
  if (!title) continue;

  const date = row['Date Published']?.trim();
  const tools = row['Tools Tested']?.trim();
  const newsletter = row['Newsletter']?.trim();
  const websiteHeader = row['Website Header']?.trim();
  const websiteSummary = row['Website Summary']?.trim();

  const slug = createSlug(title);
  const formattedDate = formatDate(date);

  // Use website summary if available, otherwise use first paragraph of newsletter
  let excerpt = websiteSummary || '';
  if (!excerpt && newsletter) {
    const firstPara = newsletter.split('\n\n')[0];
    excerpt = firstPara.substring(0, 200) + (firstPara.length > 200 ? '...' : '');
  }

  // Extract tool names for tags
  const toolTags = [];
  if (tools) {
    if (tools.includes('Claude')) toolTags.push('Claude');
    if (tools.includes('ChatGPT')) toolTags.push('ChatGPT');
    if (tools.includes('Gemini')) toolTags.push('Gemini');
    if (tools.includes('Perplexity')) toolTags.push('Perplexity');
    if (tools.includes('NotebookLM')) toolTags.push('NotebookLM');
  }

  // Create markdown content
  const content = `---
title: "${title.replace(/"/g, '\\"')}"
date: "${formattedDate}"
category: "Tool Comparison"
excerpt: "${excerpt.replace(/"/g, '\\"').replace(/\n/g, ' ')}"
image: "/images/posts/${slug}.jpg"
tags: ${JSON.stringify(toolTags)}
---

${websiteHeader ? `## ${websiteHeader}\n\n` : ''}

${newsletter || 'Content coming soon...'}

${tools ? `\n\n## Tools Used\n\n${tools}\n` : ''}
`;

  // Write to file
  const postsDir = path.join(__dirname, 'posts');
  if (!fs.existsSync(postsDir)) {
    fs.mkdirSync(postsDir, { recursive: true });
  }

  const filePath = path.join(postsDir, `${slug}.md`);
  fs.writeFileSync(filePath, content);

  postsCreated++;
  console.log(`✅ Created: ${slug}.md`);
}

console.log(`\n🎉 Successfully created ${postsCreated} blog posts!`);
console.log(`\n📸 Image placeholders created at: /images/posts/[post-name].jpg`);
console.log(`   Add your images to: public/images/posts/`);

File: import-posts.py
#!/usr/bin/env python3
import csv
import os
import re
from datetime import datetime

# Paths
csv_path = '/Users/kimberlybella/Downloads/Daily(ish) GenAI Newsletter - Sheet1.csv'
posts_dir = '/Users/kimberlybella/Library/Mobile Documents/com~apple~CloudDocs/Daily(ish) GenAI Experiments/dailyish-genai-experiments/posts'

# Create posts directory if it doesn't exist
os.makedirs(posts_dir, exist_ok=True)

def create_slug(title):
    """Create URL-friendly slug from title"""
    slug = title.lower()
    slug = re.sub(r'[^a-z0-9]+', '-', slug)
    slug = slug.strip('-')
    return slug

def format_date(date_str):
    """Convert date from M/D/YYYY to YYYY-MM-DD"""
    if not date_str:
        return datetime.now().strftime('%Y-%m-%d')

    try:
        parts = date_str.split('/')
        if len(parts) == 3:
            month, day, year = parts
            return f"{year}-{month.zfill(2)}-{day.zfill(2)}"
    except:
        pass

    return date_str

def extract_tags(tools_str):
    """Extract tool names as tags"""
    if not tools_str:
        return []

    tags = []
    tool_names = ['Claude', 'ChatGPT', 'Gemini', 'Perplexity', 'NotebookLM', 'DALL-E', 'Midjourney']

    for tool in tool_names:
        if tool.lower() in tools_str.lower():
            tags.append(tool)

    return tags

# Read CSV
posts_created = 0

with open(csv_path, 'r', encoding='utf-8') as file:
    # Use csv.reader to properly handle quoted fields
    reader = csv.DictReader(file)

    for row in reader:
        title = row.get('Title', '').strip()
        if not title:
            continue

        date = row.get('Date Published', '').strip()
        tools = row.get('Tools Tested', '').strip()
        newsletter = row.get('Newsletter', '').strip()
        website_header = row.get('Website Header', '').strip()
        website_summary = row.get('Website Summary', '').strip()

        # Create slug
        slug = create_slug(title)

        # Format date
        formatted_date = format_date(date)

        # Create excerpt
        excerpt = website_summary if website_summary else ''
        if not excerpt and newsletter:
            # Use first paragraph
            first_para = newsletter.split('\n\n')[0]
            excerpt = first_para[:200] + ('...' if len(first_para) > 200 else '')

        # Extract tags
        tags = extract_tags(tools)

        # Escape quotes in strings
        title_escaped = title.replace('"', '\\"')
        excerpt_escaped = excerpt.replace('"', '\\"').replace('\n', ' ')

        # Create markdown content
        content = f'''---
title: "{title_escaped}"
date: "{formatted_date}"
category: "Tool Comparison"
excerpt: "{excerpt_escaped}"
image: "/images/posts/{slug}.jpg"
tags: {tags}
---

{f"## {website_header}\n\n" if website_header else ""}

{newsletter if newsletter else "Content coming soon..."}

{f"\n\n## Tools Used\n\n{tools}\n" if tools else ""}
'''

        # Write to file
        file_path = os.path.join(posts_dir, f'{slug}.md')
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(content)

        posts_created += 1
        print(f"✅ Created: {slug}.md")

print(f"\n🎉 Successfully created {posts_created} blog posts!")
print(f"\n📸 Next steps:")
print(f"   1. Add your cover images to: public/images/posts/")
print(f"   2. Name them to match the post slug: [post-name].jpg")
print(f"   3. Push to GitHub and your site will update automatically!")
